lib/internal/test_runner/test.js

```diff
@@ -722,6 +722,16 @@ class Test extends AsyncResource {
       // This helps catch any asynchronous activity that occurs after the tests
       // have finished executing.
       this.postRun();
+    } else if (forceExit) {
+      // This is the root test, and all known tests and hooks have finished
+      // executing. If the user wants to force exit the process regardless of
+      // any remaining ref'ed handles, then do that now. It is theoretically
+      // possible that a ref'ed handle could asynchronously create more tests,
+      // but the user opted into this behavior.
+      this.reporter.once('close', () => {
+        process.exit();
```

Comment:
1. what if the reporter is already closed, should we check if the reporter is already closed?
2. I think we should check if there are active handles before exiting so maybe the force exit is not needed
3. maybe print a warning that some handles are refed and it's best to clean them up

Reply:
> what if the reporter is already closed

in this case, there is no need for forced exit. the reporter closes on a call to `root.postRun`